util/reggen/validate.py

```diff
@@ -245,8 +244,8 @@ def bitmask(bfield):
         msb = brange[0]
         lsb = brange[2]
         res = 0
-        if ((not msb.isdecimal()) or (not lsb.isdecimal()) or
-            (int(lsb) > int(msb))):
+        if (not msb.isdecimal()) or (not lsb.isdecimal()) or\
+           (int(lsb) > int(msb)):
```

Comment:
This also feels a bit icky. You can make the line shorter by converting !A or !B to !(A and B) and the result fits in 79 chars:

```python
        if not (msb.isdecimal() and lsb.isdecimal()) or (int(lsb) > int(msb)):
            log.error("Bad bit range " + bfield + str(brange))
            return (0, 0, 0)
```

It also seems a bit sad to repeatedly convert the strings to integers (each call to `int` is an `atoi`!). For a future commit, maybe something like this would be cleaner?

```python
def bitmask(bfield):
    if ':' in bfield:
        msb_str, lsb_str = bfield.split(':', 1)
        if not msb_str.isdecimal():
            log.error('Bad bit range: {!r} (MSB is {!r}, not an integer >= 0)'
                      .format(bfield, msb_str))
        if not lsb_str.isdecimal():
            log.error('Bad bit range: {!r} (LSB is {!r}, not an integer >= 0)'
                      .format(bfield, lsb_str))

        msb = int(msb_str)
        lsb = int(lsb_str)

        if msb < lsb:
            log.error("Bad bit range: {!r} (msb < lsb)".format(bfield))
            return (0, 0, 0)

        return ((1 << (msb + 1)) - (1 << lsb), (msb + 1) - lsb, lsb)

    if not bfield.isdecimal():
        log.error("Bad bit field: {!r} (should be a bit index or a range A:B)"
                  .format(bfield))
        return (0, 0, 0)

    idx = int(bfield)
    return (1 << idx, 1, idx)
```
I think this also improves the error messages if something goes wrong. It's a bit sad that this code doesn't raise an exception on bogus data, though. Does every call site have to check for a return value of `(0, 0, 0)`? Maybe it would be better to raise a `ValueError`?

Reply:
Actually, I just re-read what I wrote and there are some stray parens in the first suggestion. Since `or` binds looser than `>`, it's shorter to write this:
```python
        if not (msb.isdecimal() and lsb.isdecimal()) or int(lsb) > int(msb):
            log.error("Bad bit range " + bfield + str(brange))
            return (0, 0, 0)
```
(and I don't think the extra parens are needed for readability: no-one is likely to parse the expression `A or B > C` as `(A or B) > C`, I don't think).